### PR TITLE
CLI: introduced --watch flag

### LINKED ISSUE #2572: keptn get event - allow to poll cloud-events (e.g., by cloud-event-id)
Right now `keptn get event` only works for dedicated event-types, e.g.

```console
$ keptn get event evaluation-done --keptn-context=8576636f-61f7-4cdf-af81-a2899a10fc4a
CLOUD_EVENT1_CONTENT, 
CLOUD_EVENT2_CONTENT, 
...
```

This behaviour should be generic, such that we can just say

```console
$ keptn get event --keptn-context=8576636f-61f7-4cdf-af81-a2899a10fc4a
CLOUD_EVENT1_CONTENT, 
CLOUD_EVENT2_CONTENT, 
...
```

In addition, the CLI call should be extended with a `-w` flag, which indicates that we "wait" for output (polling), similar to what kubectl does, e.g.:
```console
$ kubectl -n keptn get pods -w
NAME                                     READY   STATUS    RESTARTS   AGE
api-service-c446ddcb4-hvnr2              1/1     Running   0          5d1h
bridge-5778896875-wsjp5                  1/1     Running   0          5d1h
configuration-service-7f5b678649-sg9bw   2/2     Running   0          5d1h
dynatrace-service-84fd6c4c7b-lwshz       2/2     Running   0          5d1h
dynatrace-sli-service-6bb58499c6-mpqxq   2/2     Running   0          34d
--- # At this point I killed dynatrace-sli-service-6bb58499c6-mpqxq
dynatrace-sli-service-6bb58499c6-mpqxq   2/2     Terminating   0          34d
dynatrace-sli-service-6bb58499c6-hl8bh   0/2     Pending       0          0s
dynatrace-sli-service-6bb58499c6-hl8bh   0/2     Pending       0          0s
dynatrace-sli-service-6bb58499c6-hl8bh   0/2     ContainerCreating   0          0s
dynatrace-sli-service-6bb58499c6-mpqxq   0/2     Terminating         0          34d
dynatrace-sli-service-6bb58499c6-mpqxq   0/2     Terminating         0          34d
dynatrace-sli-service-6bb58499c6-mpqxq   0/2     Terminating         0          34d
dynatrace-sli-service-6bb58499c6-hl8bh   2/2     Running             0          3s

```

`kubectl -w` does the following:   -w, --watch=false: After listing/getting the requested object, watch for changes. Uninitialized objects are excluded if no object name is provided.

For us, I believe we can introduce the `-w`/`--watch` flag in `keptn get event`, such that we can do the following:

```console
$ keptn send event new-artifact --project=ck-sockshop --service=carts --image=docker.io/keptnexamples/carts --tag=0.11.3
Starting to send a new-artifact-event to deploy the service carts in project ck-sockshop in version docker.io/keptnexamples/carts:0.11.3
ID of Keptn context: b8407f3f-1748-4812-80ee-b54783d38dd6

$ keptn get event --keptn-context=b8407f3f-1748-4812-80ee-b54783d38dd6 -w
{
  "contenttype": "application/json",
  "data": {
    "canary": {
      "action": "set",
      "value": 100
    },
    "eventContext": null,
    "labels": null,
    "project": "ck-sockshop",
    "service": "carts",
    "stage": "",
    "valuesCanary": {
      "image": "docker.io/keptnexamples/carts:0.11.3"
    }
  },
  "id": "bc0574f7-69f0-4583-918d-0a9621c54eb0",
  "source": "https://github.com/keptn/keptn/cli#configuration-change",
  "specversion": "0.2",
  "time": "2020-10-21T12:55:19.885Z",
  "type": "sh.keptn.event.configuration.change",
  "shkeptncontext": "b8407f3f-1748-4812-80ee-b54783d38dd6"
},
{
  "contenttype": "application/json",
  "data": {
    "deploymentURILocal": "http://carts.ck-sockshop-dev",
    "deploymentURIPublic": "https://carts.ck-sockshop-dev.my-host.com:443",
    "deploymentstrategy": "direct",
    "image": "docker.io/keptnexamples/carts",
    "labels": null,
    "project": "ck-sockshop",
    "service": "carts",
    "stage": "dev",
    "tag": "0.11.3",
    "teststrategy": "functional"
  },
  "id": "b7cc0df1-5e37-488a-8e37-2f9e48da81c8",
  "source": "helm-service",
  "specversion": "0.2",
  "time": "2020-10-21T12:56:32.314Z",
  "type": "sh.keptn.events.deployment-finished",
  "shkeptncontext": "b8407f3f-1748-4812-80ee-b54783d38dd6"
},
...

```
until the user hits CTRL-C or kills the process otherwise.

The same could be done for evaluation-done events, e.g.:

```console

$ keptn get event evaluation-done --keptn-context=b8407f3f-1748-4812-80ee-b54783d38dd6 -w
{
	"contenttype": "application/json",
	"data": {
		"deploymentstrategy": "direct",
		"evaluationdetails": {
			"comparedEvents": null,
			"indicatorResults": null,
			"result": "no evaluation performed by lighthouse because no SLO file configured for project ck-sockshop",
			"score": 0,
			"sloFileContent": "",
			"timeEnd": "2020-10-21T12:56:41Z",
			"timeStart": "2020-10-21T12:56:32Z"
		},
		"labels": null,
		"project": "ck-sockshop",
		"result": "pass",
		"service": "carts",
		"stage": "dev",
		"teststrategy": "functional"
	},
	"id": "384160a9-11d1-4580-82db-6c5b17763397",
	"shkeptncontext": "b8407f3f-1748-4812-80ee-b54783d38dd6",
	"source": "lighthouse-service",
	"specversion": "0.2",
	"time": "2020-10-21T12:58:49.325Z",
	"type": "sh.keptn.events.evaluation-done"
},
...
```

Please note that above output was generated using Keptn 0.7.2, the output for 0.8.0 might be different depending on the cloud-events.

In addition, this event-polling functionality could also be added in other CLI parts, such as `keptn create service`, `keptn create project`, `keptn send event start-evaluation`, `keptn send event new-artifact`, or in particular, any CLI command that retrieves a cloud-event (and was usually trying to listen to the websocket) as an alternative to the websocket connection that prints the log output.

To be discussed:
* Should `--watch` be used as a toggle, or even extended with a `--watch-time=30` to indicate that we only want to watch for 30 seconds?
* Should we add the polling to the commands listed above, or rather print a message to the user stating that they should use `keptn get event --keptn-context=... -w` to watch the event stream?

### PR DESCRIPTION
**Pull request by warber**
closes #2572 

Signed-off-by: warber <bernd.warmuth@dynatrace.com>

### CODE REVIEW

File: cli/cmd/get_event.go [hunk 1]
```diff
@@ -143,11 +137,8 @@ func init() {
...
-}
 
-func setParameterValue(value string, defaultValue string) string {
-	if len(value) == 0 {
-		return defaultValue
-	}
-	return value
+	getEventParams.Watch = getEventCmd.Flags().BoolP("watch", "w", false, "Print event stream")
+
+	getEventParams.WatchTime = getEventCmd.Flags().Int("watch-time", math.MaxInt32, "Timeout (in seconds) used for the --watch flag")
```
Comment (christian-kreuzberger-dtx) [inline cli/cmd/get_event.go:143]: is `MaxInt32` best practice for such timeouts? I think something in the order of 2-5 minutes makes more sense.

File: cli/go.mod [hunk 2]
```diff
@@ -15,7 +15,7 @@ require (
 	github.com/google/uuid v1.1.1
 	github.com/googleapis/gnostic v0.1.0 // indirect
 	github.com/hashicorp/go-version v1.2.0
-	github.com/keptn/go-utils v0.8.0-alpha.0.20210108094001-51257d664f81
+	github.com/keptn/go-utils v0.8.0-alpha.0.20210113100130-2b6cb3d71512
```
Comment (christian-kreuzberger-dtx) [inline cli/go.mod:18]: Let's wait until PR https://github.com/keptn/go-utils/pull/228 is merged, and then we'll update cli/go.mod here.

File: cli/cmd/get_event.go (hunk 1 above)
Comment (warber) [reply cli/cmd/get_event.go:143]: If you do not specify a timeout I would expect such a flag to watch forever... But that's just my expectation. I'm also happy to change it to default to a couple of minutes.
Maybe we collect a third opinion? ( @bacherfl ?)

Comment (christian-kreuzberger-dtx) [reply cli/cmd/get_event.go:143]: That's a totally valid point :+1:

Comment (bacherfl) [reply cli/cmd/get_event.go:146]: I would also expect the command to wait forever if I do not specify a timeout

Comment (christian-kreuzberger-dtx) [reply cli/cmd/get_event.go:146]: Then let's do it like that ;) Thanks Florian and Bernd!

File: cli/cmd/get_event_test.go [hunk 3]
```diff
@@ -15,18 +13,6 @@ func init() {
...
-
-	cmd := fmt.Sprintf("get event --project=%s",
-		"sockshop")
-	_, err := executeActionCommandC(cmd)
-
-	if !errorContains(err, "please provide an event type as an argument") {
-		t.Errorf("missing expected error, but got %v", err)
-	}
-}
-
```
Comment (christian-kreuzberger-dtx) [inline cli/cmd/get_event_test.go:29]: Why was this test removed? Does it no longer make sense or can we query all events without a type now?

Comment (warber) [reply cli/cmd/get_event_test.go:29]: yes, the new implementation allows passing no argument. In this case the result will just not be filtered by the event type.The test makes no sense anymore

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM

**Pull request merged**